torch/csrc/autograd/python_cpp_function.cpp

```diff
@@ -27,6 +28,12 @@ PyObject* THPCppFunction_call(PyObject* self, PyObject* args, PyObject *kwargs)
   }
 
   int num_inputs = PyTuple_GET_SIZE(args);
+  int num_inputs_required = ((THPCppFunction*)self)->cdata->num_inputs();
+  std::string self_name = ((THPCppFunction*)self)->cdata->name();
+  if ((self_name.find("Error") == std::string::npos) && (num_inputs != num_inputs_required)) {
```

Comment:
This is not a good way to determine identity of the function. You can try to `dynamic_cast` them (and check for nullptr), but it really would be better to just fix Error to take the number of inputs in its constructor.

Reply:
There are two solution based on the latter way (number of inputs in the constructor):
1. This solution might require making `num_inputs()` `virtual`. Add `input_nr` and return `input_nr` when `num_inputs()` is called.
2. This solution might require adding metadata, but without making `num_inputs()` virtual. Emplace `input_nr` times into `input_metadata` so that the return value of `num_inputs()` matches `input_nr`.

Which one do you think is better?